src/weights.jl

```diff
@@ -682,6 +682,31 @@ function mean(A::AbstractArray, w::UnitWeights; dims::Union{Colon,Int}=:)
     return mean(A, dims=dims)
 end
 
+"""
+    mean(f, A::AbstractArray, w::AbstractWeights[, dims::Int])
```

Comment:
```suggestion
    mean(f, A::AbstractArray, w::AbstractWeights[; dims])
```
`dims` shouldn't be required to be an integer.